### PR TITLE
 Add pillar and data match modules

### PR DESCRIPTION
**Pull request by brunobottazzini**
- This patch also includes documentation for the methods inside
the Match module class.
- Add unit test case for the Match module

### CODE REVIEW

File: src/test/java/com/suse/salt/netapi/calls/modules/MatchTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    private static final int MOCK_HTTP_PORT = 8888;
+
+    private SaltClient client;
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(MOCK_HTTP_PORT);
+
+    static final String JSON_CONFIG_MANAGE_MODE = ClientUtils.streamToString(
+            FileTest.class.getResourceAsStream("/modules/match/match_output.json"));
```
Comment (renner) [inline src/test/java/com/suse/salt/netapi/calls/modules/MatchTest.java]: Is there a reason why the name was chosen as `JSON_CONFIG_MANAGE_MODE `? I am only curious why it was not rather named `JSON_MATCH_OUTPUT` or similar.

File: src/main/java/com/suse/salt/netapi/calls/modules/Match.java [hunk 2]
```diff
@@ -47,6 +76,39 @@
                 Optional.of(args), new TypeToken<Boolean>(){});
     }
 
+    /**
+     * Return True if the minion matches the given pillar target.
+     * The delimiter argument can be used to specify a different delimiter.
+     * @param tgt
+     * @param delimiter
```
Comment (renner) [inline src/main/java/com/suse/salt/netapi/calls/modules/Match.java]: As you added javadoc comments I might suggest you to also add for completeness short descriptions of all the parameters as it is usually done, for example:

```java
@param tgt pillar target
@param delimiter optional different delimiter
```

Comment (brunobottazzini) [reply src/main/java/com/suse/salt/netapi/calls/modules/Match.java]: I will add it.

File: src/test/java/com/suse/salt/netapi/calls/modules/MatchTest.java (hunk 1 above)
Comment (brunobottazzini) [reply src/test/java/com/suse/salt/netapi/calls/modules/MatchTest.java]: Hello Renner, 

Sorry in the delay to answer your message. I was on Vacation :)...
You are right. I will change the name and update the PR. Today....

Thank you

File: src/main/java/com/suse/salt/netapi/calls/modules/Match.java (hunk 2 above)
Comment (brunobottazzini) [reply src/main/java/com/suse/salt/netapi/calls/modules/Match.java]: @renner, I just pushed everything... I hope it can me merged :)

**Comment by renner** [comment]
Sorry for the long delay @brunobottazzini, finally merged it.

**Pull request merged**